ui/app/components/secret-engine/configure-azure.ts

```diff
@@ -0,0 +1,181 @@
+  @service declare readonly version: VersionService;
+  @service declare readonly flashMessages: FlashMessageService;
+
+  @tracked errorMessage: string | null = null;
+  @tracked invalidFormAlert: string | null = null;
+  @tracked accessType = 'azure';
+  @tracked saveIssuerWarning = '';
+
+  disableAccessType = false;
+
+  constructor(owner: unknown, args: Args) {
+    super(owner, args);
+
+    if (this.version.isCommunity || this.args.model.isNew) return; // the following checks are relevant only to enterprise users and those editing an existing configuration.
+
+    const { identityTokenAudience, identityTokenTtl, clientSecret, rootPasswordTtl } = this.args.model;
+    const wifAttributesSet = !!identityTokenAudience || !!identityTokenTtl;
+    const azureAttributesSet = !!clientSecret || !!rootPasswordTtl;
+    // if any WIF attributes have been set in the model, set accessType to 'wif'
+    this.accessType = wifAttributesSet ? 'wif' : 'azure';
+    // if there are either WIF or azure attributes, disable user's ability to change accessType
+    this.disableAccessType = wifAttributesSet || azureAttributesSet;
+  }
+
+  @action continueSubmitForm() {
+    this.saveIssuerWarning = '';
+    this.save.perform();
+  }
+
+  // check if the issuer has been changed to show issuer modal
+  // continue saving the configuration
+  submitForm = task(
+    waitFor(async (event: Event) => {
+      event?.preventDefault();
+      this.resetErrors();
+      const { issuerConfig } = this.args;
+
+      if (issuerConfig?.hasDirtyAttributes) {
+        // if the issuer has changed show modal with warning that the config will change
+        // if the modal is shown, the user has to click confirm to continue saving
+        this.saveIssuerWarning = `You are updating the global issuer config. This will overwrite Vault's current issuer ${
+          issuerConfig.queryIssuerError ? 'if it exists ' : ''
+        }and may affect other configurations using this value. Continue?`;
+        // exit task until user confirms
+        return;
+      }
+      await this.save.perform();
+    })
+  );
+
+  save = task(
+    waitFor(async () => {
+      const { model, issuerConfig } = this.args;
+      // check if any of the model attributes have changed
+      // if no changes, transition and notify user
+      // if changes, save the model and notify user
+      // note: "backend" dirties model state so explicity ignore it here
+      const attrChanged = Object.keys(model?.changedAttributes()).some((item) => item !== 'backend');
+      const issuerAttrChanged = issuerConfig?.hasDirtyAttributes;
+      if (!attrChanged && !issuerAttrChanged) {
+        this.flashMessages.info('No changes detected.');
+        this.transition();
+        return;
+      }
+
+      const modelSaved = attrChanged ? await this.saveModel() : false;
+      const issuerSaved = issuerAttrChanged ? await this.updateIssuer() : false;
+
+      if (modelSaved || issuerSaved) {
```

Comment:
I think we want to change this to `&&` - don't we only want to transition if both are successful?